scanner/src/main/kotlin/PathScanner.kt

```diff
@@ -182,7 +182,7 @@ abstract class PathScanner(
      */
     open fun getScannerCriteria(): ScannerCriteria {
         val options = scannerConfig.options?.get(scannerName).orEmpty()
-        val minVersion = parseVersion(options[PROP_CRITERIA_MIN_VERSION]) ?: Semver(normalizeVersion(expectedVersion))
+        val minVersion = parseVersion(options[PROP_CRITERIA_MIN_VERSION]) ?: Semver(normalizeVersion(version))
```

Comment:
Just asking: Can this cause any issues if the local executable cannot be resolved and hence the version is not available? Would then the database already be queried with a wrong version before scanning of local sources fails?

Reply:
In the worst case an `IOException` could be thrown when querying the version fails if the tool is not available *and* bootstrapping failed. But I guess throwing should be fine in this case, or?

Reply:
It is just because the documentation of the `version` property says: "The actual version of the scanner, or an empty string in case of failure." If the string was empty, could it be the case that `Scanner` already starts processing before a failure is thrown when the external tool is tried to be invoked?

Reply:
The docs are simply wrong / misleading, as it depends on the *kind* of failure: If the tool could be executed but the version could not be found, and empty string is returned. If the tool could not be executed in the first place, and `IOException` is thrown.

Reply:
Thanks for clarifying.